# **MÓDULO 13**
# Projeto - Fundamentos da Descoberta de Dados

Nesse projeto trabalharemos com a base de dados de produtos de um supermercado do Chile.
A ideia é que vocês apliquem os conceitos estatísticos vistos no último módulo, mais os conceitos de visualizações de dados através de gráficos e finalizem publicando no seu github!

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import plotly.express as px

Faça a leitura dos dados do arquivo CSV:


Altere o código abaixo de acordo com seu diretório.

In [ ]:

caminho_arquivo = r"C:\Users\Isabela Amorim\Downloads\MODULO7_PROJETOFINAL_BASE_SUPERMERCADO - MODULO7_PROJETOFINAL_BASE_SUPERMERCADO (1).csv.csv"

df = pd.read_csv(caminho_arquivo, delimiter=',')

df.head(10)

title       Marca  Preco_Normal  \
0              Pack 12 un, Leche extra proteína 1 L  Loncoleche         19788   
1   Pack 12 un, Leche chocolate receta original 1 L     Soprole         18228   
2    Pack 12 un, Leche semidescremada chocolate 1 L     Soprole         18228   
3     Pack 12 un, Leche semidescremada frutilla 1 L     Soprole         18228   
4       Pack 12 un, Leche sin lactosa chocolate 1 L  Loncoleche         17988   
5        Pack 12 un, Leche sin lactosa frutilla 1 L  Loncoleche         17988   
6  Pack 12 un, Leche saborizada light chocolate 1 L  Loncoleche         17988   
7         Pack 12 un, Leche saborizada frutilla 1 L       Colun         17388   
8         Pack 12 un, Leche saborizada vainilla 1 L       Colun         17388   
9           Pack 12 un, Leche saborizada manjar 1 L       Colun         17388   

   Preco_Desconto  Preco_Anterior  Desconto Categoria  
0               0               0         0   lacteos  
1               0               0         0   lacteos  
2               0               0         0   lacteos  
3               0               0         0   lacteos  
4               0               0         0   lacteos  
5               0               0         0   lacteos  
6               0               0         0   lacteos  
7               0               0         0   lacteos  
8               0               0         0   lacteos  
9               0               0         0   lacteos

Os campos do nosso dataframe são:

**Title:** Nome do produto.


**Marca:** A marca do produto.


**Preco_Normal:** O preço em que o produto costuma ser vendido quando não há desconto.


**Preco_Desconto:** O preço vendido após o desconto ser aplicado.


**Preco_Anterior:** Preço em que era comercializado o produto antes do desconto aplicado.


**Desconto:** Total de desconto aplicado.






As colunas que aparecem com valores 0 são para os produtos onde não tivemos descontos aplicados.


As categorias estão em espanhol!

# 1 - Traga a média e a mediana dos preços - coluna Preco_Normal - por categoria de produto.
# Identifique as categorias que parecem ter um valor de média abaixo ou acima da mediana.

In [4]:
df.groupby('Categoria')['Preco_Normal'].mean().reset_index().sort_values(by='Preco_Normal',ascending=False)


Categoria  Preco_Normal
1          comidas-preparadas   3095.043478
5                     lacteos   2385.219239
2                  congelados   2108.042553
0  belleza-y-cuidado-personal   1783.556485
3                      frutas   1724.473684
6                    verduras   1343.296875
4        instantaneos-y-sopas    765.491228

In [ ]:

df.groupby('Categoria')['Preco_Normal'].median().reset_index().sort_values(by='Preco_Normal',ascending=False)

Categoria  Preco_Normal
1          comidas-preparadas        3290.0
0  belleza-y-cuidado-personal        1569.0
2                  congelados        1519.0
3                      frutas        1195.0
6                    verduras        1180.0
5                     lacteos         989.0
4        instantaneos-y-sopas         439.0

Digite aqui as categorias: lacteos, congelados, frutas,instantaneos-y-sopas.

# 2 - Traga o desvio padrão por categoria de produto.
# Qual o comportamento da média e mediana nas categorias com maior desvio?

In [ ]:

df.groupby('Categoria')['Preco_Normal'].std().reset_index().sort_values(by='Preco_Normal',ascending=False)

Categoria  Preco_Normal
5                     lacteos   3925.816164
0  belleza-y-cuidado-personal   2210.041719
2                  congelados   2111.539896
1          comidas-preparadas   2019.911428
3                      frutas   1639.151114
4        instantaneos-y-sopas   1170.232869
6                    verduras   1012.699625

Das 3 categorias com maior desvio-padrão, a categoria lacteos é a maior divergente entre a média e a mediana com aproximadamente 297% de diferença, seguidos das categorias congelados e belleza-y-cuidado-personal, onde a menor divergência é de aproximadamente 12%.

# 3 - Plot um boxplot da distribuição do Preco_Normal para a categoria que você identificou que tem o maior desvio padrão. Como é a distribuição desses dados segundo o boxplot? Você identifica muitos outliers?

In [9]:

# Filtrar apenas os dados da categoria 'lacteos'
df_lacteos = df[df['Categoria'] == 'lacteos']

# Criar o boxplot
fig = px.box(df_lacteos, y='Preco_Normal', title='Boxplot da distribuição de Preço Normal para a categoria Lacteos')
fig.update_layout(
    yaxis_title='Preço Normal',
    xaxis_title='Categoria Lacteos',  
    title_font_size=20,  
    showlegend=False)

fig.show()

Com o  gráfico Boxplot é possível ver a presença de outliers com valores bem superiores a média (989), chegando a quase 20k.

# 4 - Plote um gráfico de barras onde temos a média de descontos por categoria.

In [ ]:
df_desconto_categoria = df.groupby('Categoria')['Preco_Desconto'].mean().reset_index().sort_values(by='Preco_Desconto', ascending=False)
fig = px.bar(df_desconto_categoria, x='Categoria',y='Preco_Desconto',title='Média de Descontos por categoria',labels={'Preco_Desconto':'Preço com Desconto','Categoria':'Categoria'})
fig.show()

# 5 - Plote um gráfico de mapa interativo agrupando os dados por categoria, marca e trazendo a média de desconto.

In [ ]:
# Agrupar por categoria e marca, calculando a média de desconto
df_agrupado = df.groupby(['Categoria', 'Marca'], as_index=False)['Preco_Desconto'].mean()
# Criar um treemap
fig = px.treemap(
    df_agrupado,
    path=['Categoria', 'Marca'],  # Agrupar por categoria e marca
    values='Preco_Desconto',  # Tamanho dos retângulos baseado na média de desconto
    title='Média de Desconto por Categoria e Marca'
)

fig.show()
